commands/ConsoleCommand.go

```diff
@@ -0,0 +1,67 @@
+package commands
+
+import (
+	"fmt"
+	"geoffrey/types"
+)
+
+const CommandNotFoundCode = -100
+
+var commands []types.ConsoleCommand

```

Comment:
first glance I dont think having this being package level is desired. I can't think of an alternative at the moment but I just know this makes it harder to scale. Let's say in the future you're `Main.go` spins up multiple geoffrey bots in concurrent. Then they are sharing one `commands`.

Also its in name conflict with the package name which isn't good practice

Reply:
for the name maybe `consoleCommands`

Reply:
The more I think about it the more it seems like this would be nice as a map. Since it is a registry of sorts you can `key` by command `Name` and `value` by the struct. That way your `RunCommand` can do a lookup instead of `O(n)` search. 

Still thinking about the scope and rn seems fine at package level. If you did have multiple concurrent workers then you could just mutex lock it